File: server/src/lib/sources/baidu.ts
import axios from 'axios'
import * as cheerio from 'cheerio'

export interface BaiduItem {
  title: string
  url: string
  content: string
  author: string
  publishedAt: Date
  source: 'baidu'
}

/**
 * 百度搜索爬虫（国内中文内容补充）
 */
export async function searchBaidu(query: string): Promise<BaiduItem[]> {
  try {
    const res = await axios.get('https://www.baidu.com/s', {
      params: { wd: query, rn: 10, ie: 'utf-8' },
      timeout: 12000,
      headers: {
        'User-Agent':
          'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36',
        Accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8',
        'Accept-Language': 'zh-CN,zh;q=0.9',
        'Accept-Encoding': 'gzip, deflate',
      },
    })

    const $ = cheerio.load(res.data)
    const results: BaiduItem[] = []

    // 匹配百度自然搜索结果（.result 和 .result-op）
    $('#content_left .result, #content_left .result-op').each((i, el) => {
      if (i >= 10) return
      const $el = $(el)

      const $link = $el.find('h3 a').first()
      const title = $link.text().trim()
      const url = $link.attr('href') || ''
      // 百度使用跳转链接（baidu.com/link?url=...），直接保留原始链接
      const snippet =
        $el.find('.c-abstract').first().text().trim() ||
        $el.find('[class*="abstract"]').first().text().trim() ||
        $el.find('.content-right').first().text().trim()

      if (!title || !url.startsWith('http')) return

      results.push({
        title,
        url,
        content: snippet || title,
        author: '',
        publishedAt: new Date(),
        source: 'baidu' as const,
      })
    })

    return results
  } catch (err) {
    console.error('[Baidu] search error:', (err as any)?.message)
    return []
  }
}

File: .cursor/skills/hot-monitor/scripts/lib/baidu.py
from __future__ import annotations

from datetime import datetime, timezone

import requests
from bs4 import BeautifulSoup

UA = {
    "User-Agent": (
        "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 "
        "(KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36"
    ),
    "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8",
    "Accept-Language": "zh-CN,zh;q=0.9",
}


def search(query: str) -> list[dict]:
    try:
        r = requests.get(
            "https://www.baidu.com/s",
            params={"wd": query, "rn": 10, "ie": "utf-8"},
            headers=UA,
            timeout=12,
        )
        r.raise_for_status()
        soup = BeautifulSoup(r.text, "html.parser")
        results: list[dict] = []
        for i, el in enumerate(soup.select("#content_left .result, #content_left .result-op")):
            if i >= 10:
                break
            link = el.select_one("h3 a")
            if not link:
                continue
            title = link.get_text(strip=True)
            url = link.get("href") or ""
            snippet_el = (
                el.select_one(".c-abstract")
                or el.select_one('[class*="abstract"]')
                or el.select_one(".content-right")
            )
            snippet = snippet_el.get_text(strip=True) if snippet_el else title
            if title and url.startswith("http"):
                results.append(
                    {
                        "title": title,
                        "url": url,
                        "content": snippet,
                        "author": "",
                        "publishedAt": datetime.now(timezone.utc),
                        "source": "baidu",
                    }
                )
        return results
    except Exception:
        return []
